auslib/test/admin/views/test_rules.py

```diff
@@ -594,3 +623,335 @@ def testGetRules(self):
+        self.assertEquals(ret.status_code, 400, msg=ret.data)
+        self.assertIn("Outdated Data Version", ret.data)
+
+    def testDeleteScheduledChangeWithoutPermission(self):
+        ret = self._delete("/scheduled_changes/rules/1", username="rex", qs=dict(data_version=1))
+        self.assertEquals(ret.status_code, 403, msg=ret.data)
+
+    def testDeleteNonExistentScheduledChange(self):
+        ret = self._delete("/scheduled_changes/rules/5", qs=dict(data_version=1))
+        self.assertEquals(ret.status_code, 404, msg=ret.data)
+
+    def testEnactScheduledChangeExistingRule(self):
+        ret = self._post("/scheduled_changes/rules/1/enact", username="mary")
+        self.assertEquals(ret.status_code, 200, ret.data)
+
+        sc_row = dbo.rules.scheduled_changes.t.select().where(dbo.rules.scheduled_changes.sc_id == 1).execute().fetchall()[0]
+        self.assertEquals(sc_row["complete"], True)
+
+        row = dbo.rules.t.select().where(dbo.rules.rule_id == 1).execute().fetchall()[0]
+        expected = {
+            "rule_id": 1, "priority": 100, "version": "3.5", "buildTarget": "d", "backgroundRate": 100, "mapping": "b",
+            "update_type": "minor", "data_version": 2, "alias": None, "product": None, "channel": None, "buildID": None,
+            "locale": None, "osVersion": None, "distribution": None, "distVersion": None, "headerArchitecture": None,
+            "comment": None, "whitelist": None, "systemCapabilities": None,
+        }
+        self.assertEquals(dict(row), expected)
+
+    def testEnactScheduledChangeNewRule(self):
+        ret = self._post("/scheduled_changes/rules/2/enact", username="mary")
+        self.assertEquals(ret.status_code, 200, ret.data)
+
+        sc_row = dbo.rules.scheduled_changes.t.select().where(dbo.rules.scheduled_changes.sc_id == 2).execute().fetchall()[0]
+        self.assertEquals(sc_row["complete"], True)
+
+        row = dbo.rules.t.select().where(dbo.rules.rule_id == 6).execute().fetchall()[0]
+        expected = {
+            "rule_id": 6, "priority": 50, "version": None, "buildTarget": None, "backgroundRate": 100, "mapping": "ab",
+            "update_type": "minor", "data_version": 1, "alias": None, "product": "baz", "channel": None, "buildID": None,
+            "locale": None, "osVersion": None, "distribution": None, "distVersion": None, "headerArchitecture": None,
+            "comment": None, "whitelist": None, "systemCapabilities": None,
+        }
+        self.assertEquals(dict(row), expected)
+
+    def testEnactScheduledChangeNoPermissions(self):
+        ret = self._post("/scheduled_changes/rules/2/enact", username="bob")
+        self.assertEquals(ret.status_code, 403, ret.data)
+
+    def testGetScheduledChangeHistoryRevisions(self):
+        ret = self._get("/scheduled_changes/rules/3/revisions")
+        self.assertEquals(ret.status_code, 200)
+        expected = {
+            "count": 2,
+            # TODO: should these have base_ on them ?
```

Comment:
Presumably destined for the UI, so better to be consistent with what the human expects to see for rules.